reproschema/models/tests/test_item.py

```diff
@@ -0,0 +1,301 @@
+tested both with:
+- only one response allowed
+- multiple responses allowed
+"""
+
+
+def test_radio():
+
+    response_options = ResponseOption(multipleChoice=False)
+    response_options.add_choice(name="Not at all", value=0)
+    response_options.add_choice(name="Several days", value=1)
+
+    item = Item(
+        name="radio",
+        input_type="radio",
+        question="question for radio item",
+        output_dir=item_dir,
+    )
+    item.set_input_type(response_options=response_options)
+
+    item.write()
+    item_content, expected = load_jsons(item_dir, item)
+    assert item_content == expected
+
+    clean_up(item_dir, item)
+
+    item.set_filename("radio multiple")
+    item.description = "radio multiple"
+    item.update()
+    item.set_pref_label("radio multiple")
+    item.set_question("question for radio item with multiple responses")
+    response_options.multipleChoice = True
+    item.set_input_type(response_options=response_options)
+    item.write()
+
+    item_content, expected = load_jsons(item_dir, item)
+    assert item_content == expected
+
+    clean_up(item_dir, item)
+
+
+def test_select():
+
+    response_options = ResponseOption(multipleChoice=False)
+    response_options.add_choice(name="Response option 1", value=0)
+    response_options.add_choice(name="Response option 2", value=1)
+    response_options.add_choice(name="Response option 3", value=2)
+
+    item = Item(
+        name="select",
+        input_type="select",
+        question="question for select item",
+        output_dir=item_dir,
+    )
+    item.set_input_type(response_options=response_options)
+
+    item.write()
+    item_content, expected = load_jsons(item_dir, item)
+    assert item_content == expected
+
+    clean_up(item_dir, item)
+
+    item.set_filename("select multiple")
+    item.description = "select multiple"
+    item.update()
+    item.set_pref_label("select multiple")
+    item.set_question("question for select item with multiple responses")
+    response_options.multipleChoice = True
+    item.set_input_type(response_options=response_options)
+    item.write()
+
+    item_content, expected = load_jsons(item_dir, item)
+    assert item_content == expected
+
+    clean_up(item_dir, item)
+
+
+def test_slider():
+
+    response_options = ResponseOption()
+    response_options.add_choice(name="not at all", value=0)
+    response_options.add_choice(name="a bit", value=1)
+    response_options.add_choice(name="so so", value=2)
+    response_options.add_choice(name="a lot", value=3)
+    response_options.add_choice(name="very much", value=4)
+
+    item = Item(
+        name="slider",
+        input_type="slider",
+        question="question for slider item",
+        output_dir=item_dir,
+    )
+    item.set_input_type(response_options)
+
+    item.write()
+    item_content, expected = load_jsons(item_dir, item)
+    assert item_content == expected
+
+    clean_up(item_dir, item)
+
+
+def test_item1():
```

Comment:
Note that the creation of some files can still be quite involved if they are complex: I still would like to make the API more intuitive but I am not sure how much more I can do for now.